webapp/src/components/permissions/boardPermissionGate.tsx

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react'
+
+import {useAppSelector} from '../../store/hooks'
+import {getCurrentBoard} from '../../store/boards'
+import {getCurrentTeam} from '../../store/teams'
+import {useHasPermissions} from '../../hooks/permissions'
+
+type Props = {
+    boardId?: string
+    teamId?: string
+    permissions: string[]
+    invert?: boolean
+    children: React.ReactNode
+}
+
+const BoardPermissionGate = React.memo((props: Props): React.ReactElement|null => {
+    const currentTeam = useAppSelector(getCurrentTeam)
+    const currentBoard = useAppSelector(getCurrentBoard)
+
+    const boardId = props.boardId || currentBoard?.id || ''
+    const teamId = props.teamId || currentTeam?.id || ''
+
+    const allowed = useHasPermissions(teamId, boardId, props.permissions)
+
+    if (allowed) {
+        if (props.invert) {
```

Comment:
The whole point of the invert propertie is to invert the "permission gate".

The permission gate works this way "If you have at least one of the permissions in the list of permission, you can see the content", with the invert we switch that behavior to "If you don't have any of the permissions in the list, you can see the content".

This is helpful when you want show "X" when you have the permission and "Y" when you don't have it. For example, imagine something like:

```js
<BoardPermissionGate permissions={['share_board']}><button>Share</button></BoardPermissionGate><BoardPermissionGate permissions={['share_board']} invert={true}><button>Board Members</button></BoardPermissionGate>
```

This would show you a "Share" button when you have permissions to share, and a "Board Members" board members button when you don't have that permission.

Reply:
Thanks. Based on that, I think the code is wrong for the case where `allowed == false && props.invert == true` that case will `return  null` but should return `props.children`.